test/plugin/test_out_forward.rb

```diff
@@ -621,6 +624,105 @@ def try_write(chunk)
     assert_equal(['test', time, records[1]], events[1])
   end
 
+  data('ack true' => true,
+       'ack false' => false)
+  test 'TLS transport client cert chain' do |ack|
+    omit "TLS and 'ack false' always fails. Need to debug" if !ack
+
+    # Cert generation based on
+    # https://ruby-doc.org/stdlib-2.4.0/libdoc/openssl/rdoc/OpenSSL/X509/Certificate.html
+    # examples
+    root = create_certificate("root-ca", 1)
+    root_ca = root[0]
+
+    root_ca_file = Tempfile.new("root-cert")
+    root_ca_file.write(root_ca.to_pem())
+    root_ca_file.close
+    # No need to write root key to file
+
+    intermediate = create_certificate("intermediate-ca", 2, root)
+    # The intermediate_ca is not written to a file - it will be in the
+    # client and service certificates' chain.
+
+    # Test both (root->cert) and (root->intermediate->cert)
+    parents = [root, intermediate]
+    parents.each do |parent|
+      client = create_certificate("client-cert", 3, parent, false)
```

Comment:
you can use following code:

```
client_cert, client_key = create_certificate("client-cert", 3, parent, false)
```

Reply:
oh great - new to ruby ha, didn't know it could auto-unpack